src/ring_theory/non_zero_divisors.lean

```diff
@@ -32,8 +32,8 @@ def non_zero_divisors (R : Type*) [monoid_with_zero R] : submonoid R :=
 
 localized "notation R`⁰`:9000 := non_zero_divisors R" in non_zero_divisors
 
-variables {M M' M₁ R R' : Type*} [monoid_with_zero M] [monoid_with_zero M']
-  [comm_monoid_with_zero M₁] [ring R] [comm_ring R']
+variables {M M' M₁ R R' F : Type*} [monoid_with_zero M] [monoid_with_zero M']
+  [comm_monoid_with_zero M₁] [ring R] [comm_ring R'] [monoid_with_zero_hom_class F M M']
```

Comment:
the first two lemmas only require `zero_hom_class`, so you can generalize a tiny bit